pyccel/codegen/printing/pycode.py

```diff
@@ -119,14 +120,21 @@ def _find_functional_expr_and_iterables(self, expr):
         body = expr.loops[1]
         while not isinstance(body, Assign):
             if isinstance(body, CodeBlock):
-                body = body.body
+                body = list(body.body)
+                while isinstance(body[0], FunctionalFor):
+                    func_for = body.pop(0)
+                    for b in body:
+                        b.substitute(func_for.lhs, func_for)
                 if len(body) > 1:
                     if any(not(isinstance(b, Assign) and b.lhs is dummy_var) for b in body[1:]):
                         raise NotImplementedError("Pyccel has introduced unnecessary statements which it cannot yet disambiguate in the python printer")
                 body = body[0]
```

Comment:
Can you give me an example where the body of a `FunctionalFor` instance is a `CodeBlock`?

Reply:
This happens in nested generators (`GeneratorComprehension` inherits from `FunctionalFor`). The `CodeBlock` is created at the semantic stage to contain the loop and the initialisation (e.g. `=inf` before running a `FunctionalMin`, `=0` before running a  `FunctionalSum` etc)